### PR TITLE
Add new resource iam_projects_policy_binding. addresses https://github.com/hashicorp/terraform-provider-google/issues/20198

### PR DESCRIPTION
**Pull request by derekchu-google**
```release-note:new-resource
`google_iam_projects_policy_binding` (beta)
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/iam3/resource_iam_projects_policy_binding_test.go.tmpl [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	return acctest.Nprintf(`
+resource "google_iam_principal_access_boundary_policy" "pab_policy" {
+  provider = google-beta
+  organization   = "%{org_id}"
+  location       = "global"
+  display_name   = "test project binding%{random_suffix}"
+  principal_access_boundary_policy_id = "tf-test-my-pab-policy%{random_suffix}"
+}
+
+data "google_project" "project" {}
```
Comment (priyankadeo-g) [inline mmv1/third_party/terraform/services/iam3/resource_iam_projects_policy_binding_test.go.tmpl:85]: Is this required?

Comment (derekchu-google) [reply mmv1/third_party/terraform/services/iam3/resource_iam_projects_policy_binding_test.go.tmpl]: talked about this offline. this uses the existing project

**Comment by derekchu-google** [comment]
hi there @rileykarson, this PR is ready to go. please take a look

File: mmv1/products/iam3/FoldersPolicyBinding.yaml [hunk 2]
```diff
@@ -47,6 +47,7 @@ async:
   error:
     path: 'error'
     message: 'message'
+  include_project: true
```
Comment (rileykarson) [inline mmv1/products/iam3/FoldersPolicyBinding.yaml:50]: This doesn't break anything but doesn't entirely do the right thing- it should be using the quota project, `billingProject`, but instead uses a blank `project` value (i.e. no quota project).

What did this fix for you? I'm having trouble understanding why it was needed.

File: mmv1/products/iam3/ProjectsPolicyBinding.yaml [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    description: |
+      The name of the policy binding in the format `{binding_parent/locations/{location}/policyBindings/{policy_binding_id}`
+    output: true
+  - name: 'uid'
+    type: String
+    description: |
+      Output only. The globally unique ID of the policy binding. Assigned when the policy binding is created.
+    output: true
+  - name: 'etag'
+    type: String
```
Comment (rileykarson) [inline mmv1/products/iam3/ProjectsPolicyBinding.yaml:87]: We should omit this or make it the (unfortunately underdocumented...) Fingerprint type: https://github.com/GoogleCloudPlatform/magic-modules/blob/d5e86ba02cb336a9d7819510a0b265de804c8757/mmv1/api/type.go#L705

File: mmv1/third_party/terraform/services/iam3/resource_iam_projects_policy_binding_test.go.tmpl [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+				Config: testAccIAM3ProjectsPolicyBinding_iamProjectsPolicyBindingExample_full(context),
+			},
+			{
+				ResourceName:            "google_iam_projects_policy_binding.my-project-binding",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"annotations", "location", "policy_binding_id"},
+			},
+			{
+				Config: testAccIAM3ProjectsPolicyBinding_iamProjectsPolicyBindingExample_update(context),
```
Comment (rileykarson) [inline mmv1/third_party/terraform/services/iam3/resource_iam_projects_policy_binding_test.go.tmpl:36]: Could you add a third step where we go back to `testAccIAM3ProjectsPolicyBinding_iamProjectsPolicyBindingExample_full` afterwards?

File: mmv1/products/iam3/FoldersPolicyBinding.yaml (hunk 2 above)
Comment (derekchu-google) [reply mmv1/products/iam3/FoldersPolicyBinding.yaml:50]: i worked with @slevenick quite a bit on this one. basically there are 3 resources, folder_policy_binding, organizations_policy_binding, projects_policy_binding that all bind to a folder/org/project to a principal_access_policy resource. to use the project (versus creating a new one), this was necessary to ensure the tests pass for folders and orgs.

File: mmv1/products/iam3/ProjectsPolicyBinding.yaml (hunk 3 above)
Comment (derekchu-google) [reply mmv1/products/iam3/ProjectsPolicyBinding.yaml:87]: good to know. will do

File: mmv1/products/iam3/FoldersPolicyBinding.yaml (hunk 2 above)
Comment (rileykarson) [reply mmv1/products/iam3/FoldersPolicyBinding.yaml:50]: Got it!

File: mmv1/third_party/terraform/services/iam3/resource_iam_projects_policy_binding_test.go.tmpl (hunk 4 above)
Comment (derekchu-google) [reply mmv1/third_party/terraform/services/iam3/resource_iam_projects_policy_binding_test.go.tmpl:36]: will do

**Review by rileykarson — APPROVED** [review]

**Pull request merged**